WasatchNET/Driver.cs

```diff
@@ -353,36 +397,27 @@ void logDevice(UsbRegistry usbRegistry)
                         logNameValuePairs(endpointInfo.ToString().Split(new char[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries), "    ");
                     }
                 }
+            }
+            device.Close();
 
-                // log SymbolicName 
-                UsbSymbolicName symName = UsbSymbolicName.Parse(usbRegistry.SymbolicName);
-                logger.debug("Symbolic Name:");
-                logger.debug("  VID:          0x{0:x4}", symName.Vid);
-                logger.debug("  PID:          0x{0:x4}", symName.Pid);
-                logger.debug("  SerialNumber: {0}", symName.SerialNumber);
-                logger.debug("  Class Guid:   {0}", symName.ClassGuid);
-
-                logger.debug("Device Properties:");
-                foreach (KeyValuePair<string, object> pair in usbRegistry.DeviceProperties)
-                {
-                    string key = pair.Key;
-                    object value = pair.Value;
+            UsbSymbolicName symName = UsbSymbolicName.Parse(usbRegistry.SymbolicName);
+            logger.debug("Symbolic Name:");
+            logger.debug("  VID:          0x{0:x4}", symName.Vid);
+            logger.debug("  PID:          0x{0:x4}", symName.Pid);
+            logger.debug("  SerialNumber: {0}", symName.SerialNumber);
+            logger.debug("  Class Guid:   {0}", symName.ClassGuid);
 
-                    // handle array values
-                    if (value is string[])
-                    {
-                        string[] values = value as string[];
-                        logger.debug("  {0}: [ {1} ]", key, string.Join(", ", values));
-                    }
-                    else
-                    {
-                        logger.debug("  {0}: {1}", key, value);
-                    }
-                }
+            logger.debug("Device Properties:");
+            foreach (KeyValuePair<string, object> pair in usbRegistry.DeviceProperties)
+            {
+                string key = pair.Key;
+                object value = pair.Value;
 
-                logger.debug(" ");
+                if (value is string[])
+                    logger.debug("  {0}: {1}", key, string.Format("[ {0} ]", string.Join(", ", value as string[])));
+                else if (value is string)
+                    logger.debug("  {0}: {1}", key, value as string);
             }
-            device.Close();

```

Comment:
In that particular line, I moved the device.Close() up a bit, so the resource was disposed as soon as it was no longer needed (the remaining code didn't actually use the device object).  Nothing really changed in this method functionally...I tightened it up a little because for awhile, I was using it to "generateUniqueKey", before I moved that functionality inside Spectrometer for better encapsulation.